app/controllers/ChoiceController.scala

```diff
@@ -54,16 +54,19 @@ class ChoiceController @Inject()(
       .fold(
         (formWithErrors: Form[Choice]) => Future.successful(BadRequest(choice_page(formWithErrors))),
         validChoice =>
-          customsCacheService.cache[Choice](eoriCacheId, choiceId, validChoice).map { _ =>
-            validChoice.value match {
-              case SupplementaryDec | StandardDec =>
-                Redirect(controllers.declaration.routes.DispatchLocationPageController.displayPage())
-              case CancelDec =>
-                Redirect(controllers.routes.CancelDeclarationController.displayForm())
-              case Submissions =>
-                Redirect(controllers.routes.NotificationsController.listOfSubmissions())
-              case _ =>
-                Redirect(controllers.routes.ChoiceController.displayChoiceForm())
+          customsCacheService.cache[Choice](eoriCacheId, choiceId, validChoice).flatMap { _ =>
+            customsCacheService.cache[Choice](s"${validChoice.value}-${request.user.eori}", choiceId, validChoice).map {
```

Comment:
just want to check why is this required? why can't you get choice from journeyAction, for each request the choice exists for the method to pass to services to use jourenytype I believe

Reply:
This is ChoiceController, so you don't know what is the user choice.

But I'm not sure that I understand idea to save choice two times in different caches.

Is it a part of migration? and when you change retrieving Choice from different place you'll remove one of the method. In other case I don't see any specific reason to do it in this way.

Reply:
Maybe I am missing something, but what is the point in storing choice value in 2 places? What is more, will we not end up with 3 documents in mongo when the user is making any declaration?

Reply:
I have removed this :) Please review again.